### PR TITLE
Infra improvements

### PR DESCRIPTION
**Pull request by suzhoum**
Issue #, if available:

Description of changes:
1. Removes `--dev-branch` option, the deployment now looks at the local installed `autogluon.bench` version (PyPI or dev) to determine which `autogluon.bench` to install in benchmark environment: either pull from PyPI, or copy the local project source to the Docker context for installation.
2. Reduces docker build time by using unified base directory for environment setup to increase chance of caching
3. Submit AWS Array Job instead of individual jobs when number of jobs is more than 1 to avoid getting throttled on AWS
4. Support both custom metrics and standard metrics in the same set of datasets for multimodal
5. Batch instance IMDSV2 compliance
6. Bug fixes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/bench/runbenchmark.py [hunk 1]
```diff
@@ -191,8 +194,13 @@ def get_job_status(
 
     for job_id in job_ids:
         response = batch_client.describe_jobs(jobs=[job_id])
-        job = response["jobs"][0]
-        status_dict[job_id] = job["status"]
+        job_detail = response["jobs"][0]
+
+        # Check if the job is an array job
```
Comment (prateekdesai04) [inline src/autogluon/bench/runbenchmark.py:199]: If the job which is submitted is more than 1 than is it not going to be an Array job by default?

File: src/autogluon/bench/runbenchmark.py [hunk 2]
```diff
@@ -205,42 +213,47 @@ def wait_for_jobs(
...
+                            all_jobs_completed = False
         except botocore.exceptions.ClientError as e:
             logger.error(f"An error occurred: {e}.")
             return
 
         if all_jobs_completed:
             break
         else:
-            time.sleep(frequency)  # Poll job statuses every 60 seconds
+            time.sleep(frequency)  # Poll job statuses every 120 seconds
```
Comment (prateekdesai04) [inline src/autogluon/bench/runbenchmark.py:252]: why do we increase the poll time?

**Comment by prateekdesai04** [comment]
A question: does this PR in any way change the format of how we construct the cloud configs to run benchmarks?

File: src/autogluon/bench/runbenchmark.py (hunk 1 above)
Comment (suzhoum) [reply src/autogluon/bench/runbenchmark.py:199]: Array job is a single parent job. Technically at this moment, `job_ids` array will contain only one object.

File: src/autogluon/bench/runbenchmark.py (hunk 2 above)
Comment (suzhoum) [reply src/autogluon/bench/runbenchmark.py:252]: It's been 120s for a while but I forgot to update the comment.

**Comment by suzhoum** [comment]
> A question: does this PR in any way change the format of how we construct the cloud configs to run benchmarks?

No. The only API change is that we removed `--dev-branch`

**Review by prateekdesai04 — APPROVED** [review]
LGTM!

**Pull request merged**